File: src/main/java/com/werfad/utils/StringUtils.java
package com.werfad.utils;

import java.util.ArrayList;
import java.util.List;

public class StringUtils {
    public static List<Integer> findAll(String s, char c) {
        List<Integer> res = new ArrayList<>();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == c) {
                res.add(i);
            }
        }
        return res;
    }

    public static List<Integer> findAllIgnoreCase(String s, char c) {
        List<Integer> res = new ArrayList<>();
        char upper = Character.toUpperCase(c);
        char lower = Character.toLowerCase(c);

        for (int i = 0; i < s.length(); i++) {
            char charAt = s.charAt(i);
            if (charAt == upper || charAt == lower) {
                res.add(i);
            }
        }

        return res;
    }
}

File: src/main/kotlin/com/werfad/utils/StringUtils.kt
package com.werfad.utils

fun String.findAll(c: Char, ignoreCase: Boolean = false): List<Int> {
    val res = ArrayList<Int>()
    var index = indexOf(c, ignoreCase = ignoreCase)
    while (index >= 0) {
        res.add(index)
        index = indexOf(c, index + 1, ignoreCase)
    }
    return res
}

fun String.findAll(find: String, ignoreCase: Boolean = false): List<Int> {
    val res = ArrayList<Int>()
    var index = indexOf(find, ignoreCase = ignoreCase)
    while (index >= 0) {
        res.add(index)
        index = indexOf(find, index + 1, ignoreCase)
    }
    return res
}
